pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -442,6 +458,28 @@ private IdealState addLLCRealtimeSegmentsInIdealState(final IdealState idealStat
     return idealState;
   }
 
+  private SegmentPartitionMetadata getPartitionMetadataFromTableConfig(String tableName, int partitionNum, int partitionId) {
```

Comment:
Confusing variable naming here. What is partitionNum? If it is number of partitions (I think that is it), then please rename it as numPartitions. Otherwise, it is unclear what the difference is between partitionNum and partitionId